src/opm/parser/eclipse/EclipseState/Aquancon.cpp

```diff
@@ -55,37 +52,44 @@ namespace Opm {
         aqurecords.resize(aquanconKeyword.size());
         aquiferID_per_record.resize(aquanconKeyword.size());
 
-        // We now do a loop over each record entry in aquancon
         for (size_t aquanconRecordIdx = 0; aquanconRecordIdx < aquanconKeyword.size(); ++aquanconRecordIdx)
         {
             const auto& aquanconRecord = aquanconKeyword.getRecord(aquanconRecordIdx);
             aquiferID_per_record.at(aquanconRecordIdx) = aquanconRecord.getItem("AQUIFER_ID").template get<int>(0);
 
-            aqurecords.at(aquanconRecordIdx).i1 = aquanconRecord.getItem("I1").template get<int>(0);
-            aqurecords.at(aquanconRecordIdx).i2 = aquanconRecord.getItem("I2").template get<int>(0);
-            aqurecords.at(aquanconRecordIdx).j1 = aquanconRecord.getItem("J1").template get<int>(0);
-            aqurecords.at(aquanconRecordIdx).j2 = aquanconRecord.getItem("J2").template get<int>(0);
-            aqurecords.at(aquanconRecordIdx).k1 = aquanconRecord.getItem("K1").template get<int>(0);
-            aqurecords.at(aquanconRecordIdx).k2 = aquanconRecord.getItem("K2").template get<int>(0);
+            // offset the indices
+            const int i1 = aquanconRecord.getItem("I1").template get<int>(0) - 1;
+            const int i2 = aquanconRecord.getItem("I2").template get<int>(0) - 1;
+            const int j1 = aquanconRecord.getItem("J1").template get<int>(0) - 1;
+            const int j2 = aquanconRecord.getItem("J2").template get<int>(0) - 1;
+            const int k1 = aquanconRecord.getItem("K1").template get<int>(0) - 1;
+            const int k2 = aquanconRecord.getItem("K2").template get<int>(0) - 1;
 
-            aquiferID_per_record.at(aquanconRecordIdx) = aquanconRecord.getItem("AQUIFER_ID").template get<int>(0);
             m_maxAquID = (m_maxAquID < aquiferID_per_record.at(aquanconRecordIdx) )?
                             aquiferID_per_record.at(aquanconRecordIdx) : m_maxAquID;
 
-            double influx_mult = aquanconRecord.getItem("INFLUX_MULT").getSIDouble(0);
+            const FaceDir::DirEnum faceDir = FaceDir::FromString(aquanconRecord.getItem("FACE").getTrimmedString(0));
 
-            FaceDir::DirEnum faceDir = FaceDir::FromString(aquanconRecord.getItem("FACE").getTrimmedString(0));
+            // whether allow aquifer connections to locate inside reservoir
+            const std::string& str_inside_reservoir  = aquanconRecord.getItem("CONNECT_ADJOINING_ACTIVE_CELL").getTrimmedString(0);
+            // not sure whether we should give a warning when input other than "YES" or "NO"
+            const bool allow_aquifer_inside_reservoir = str_inside_reservoir == "YES" ? true : false;
```

Comment:
Here you could use `DeckItem::to_bool( )`

Reply:
Thanks for this comment. But `to_bool()` has a much more possibilities (like Y, 1, N, 0) to be true or false. Here, by default it is `NO`, which is false. I want to make enforce only when you specify `YES`, it will be true. For all the other situations, it is false. 

Not sure how E100 handles illegal input. Sometimes it uses the default values when the input is not legal. I could not test.  If finally testing show differently, we can update here.